mPower2/mPower2/PassiveCollectors.swift

```diff
@@ -754,11 +780,40 @@ class PassiveGaitCollector : NSObject, PassiveLocationTriggeredCollector {
         let geofence = CLCircularRegion(center: validLocation.coordinate, radius: regionRadius, identifier: kPassiveGaitRegionIdentifier)
         self.locationManager?.startMonitoring(for: geofence)
         
+        if let lastLocation = previousGeofenceLocation {
+            let dataPoint = RSDDistanceRecord(uptime: ProcessInfo.processInfo.systemUptime,
+                                              timestamp: 0,
+                                              stepPath: "Passive",
+                                              location: validLocation,
+                                              previousLocation: lastLocation,
+                                              totalDistance: nil,
+                                              relativeDistanceOnly: true)
+            uploadDisplacement(dataPoint)
+        }
+        previousGeofenceLocation = validLocation
+
         // End the background task that was keeping us alive to get here.
         self.endBackgroundPausingTask()
 
     }
     
+    /// Upload a PassiveDisplacementGeofence record.
+    private func uploadDisplacement(_ displacementData: RSDDistanceRecord) {
+        let archiveFilename = kPassiveDisplacementArchiveFilename
+        let archive = SBBDataArchive(reference: kPassiveDisplacementGeofenceSchemaID, jsonValidationMapping: nil)
+        if let schemaRevision = MP2BridgeConfiguration.shared.schemaInfo(for: kPassiveDisplacementGeofenceSchemaID)?.schemaVersion {
+            archive.setArchiveInfoObject(schemaRevision, forKey: kSchemaRevisionKey)
+        }
+
+        do {
+            let data = try displacementData.rsd_jsonEncodedData()
+            archive.insertData(intoArchive: data, filename: archiveFilename, createdOn: Date())
+            try archive.complete()
+            archive.encryptAndUploadArchive()
+        }
+        catch {}
```

Comment:
Should probably have a print statement with a warning that upload failed.

Reply:
good point.